packages/core/src/components/fields/SchemaField.tsx

```diff
@@ -231,9 +231,10 @@ function SchemaFieldRender<T = any, S extends StrictRJSFSchema = RJSFSchema, F e
   );
   /*
    * AnyOf/OneOf errors handled by child schema
+   * unless it can be rendered as select control
    */
   const errorsComponent =
-    hideError || schema.anyOf || schema.oneOf ? undefined : (
+    hideError || ((schema.anyOf || schema.oneOf) && !schemaUtils.isSelect(schema)) ? undefined : (
```

Comment:
This seems right to me. I think this select case is the only place where we want to handle the error in the parent rather than the children.